hooks/useLocal.tsx

```diff
@@ -0,0 +1,161 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { MonthConstant } from '@constants';
+
+export interface PathData {
+  pathId: number;
+  saveData: { angNumber: number; verseId: number };
+  progress: number;
+  startDate: string;
+  completionDate: string;
+  pathName: string;
+}
+export interface DateData {
+  pathid: number;
+  dates: PathDate[];
+  scrollPosition: number;
+}
+export interface PathDate {
+  date: string;
+  angs: number;
+}
+export interface FontSizeData {
+  fontSize: string;
+  number: number;
+}
+export interface AngsFormat {
+  format: string;
+}
+export const useLocal = () => {
+  const fetchFromLocal = async () => {
+    const pathFromLocal = await AsyncStorage.getItem('pathDetails');
+    const pathFromLocalArray: PathData[] = pathFromLocal ? JSON.parse(pathFromLocal) : [];
+    const pathDateData = await AsyncStorage.getItem('pathDateDetails');
+    const pathDateDataArray: DateData[] = pathDateData ? JSON.parse(pathDateData) : [];
+    return { pathDataArray: pathFromLocalArray, pathDateDataArray: pathDateDataArray };
+  };
```

Comment:
This doesn't handle JSON parsing errors. If the stored data is corrupted, this will crash the app.

```javascript
const fetchFromLocal = async () => {
  try {
    const pathFromLocal = await AsyncStorage.getItem('pathDetails');
    const pathFromLocalArray: PathData[] = pathFromLocal ? JSON.parse(pathFromLocal) : [];
    // ... rest of the code
  } catch (error) {
    console.error('Error parsing stored data:', error);
    return { pathDataArray: [], pathDateDataArray: [] };
  }
};
```